build.sbt

```diff
@@ -15,7 +15,11 @@ libraryDependencies ++= Seq(
     "com.typesafe" % "config" % "1.4.2" % Test
 )
 
-ThisBuild / scalacOptions ++= Seq("-Xsource:3", "-deprecation", "-feature", "-language:postfixOps")
+ThisBuild / scalacOptions ++= Seq("-deprecation", "-feature", "-language:postfixOps") ++
+  (CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, _)) => Seq("-Xsource:3") // flags only needed in Scala 2
+    case _ => Seq.empty
+  })
```

Comment:
Scala 3 doesn't have the `"-Xsource:3"` flag - because it _is_ Scala 3, with Scala 3 syntax, and isn't doing the thing `"-Xsource:3"` activates in Scala 2, _adopting_ Scala 3 syntax.

Consequently, we get a compiler warning if we use `"-Xsource:3"` in Scala 3, which we avoid here by only using the flag in Scala 2.